### PR TITLE
Adding ReactInit and ReactFunction

### PR DESCRIPTION
**Pull request by chrisglein**
Could be more documentation based on the content of [NativeModules.h](https://github.com/microsoft/react-native-windows/blob/0dfe9decb4390c4b77e0837e69c4a25878034960/vnext/Microsoft.ReactNative.Cxx/NativeModules.h), but followed the pattern of others to just put a quick sentence. 

Added on all versions back to 0.62, which is when these were added.

### CODE REVIEW

**Comment by chrisglein** [comment]
Fixes #254
(unless we want to break out pages with more details)

File: docs/native-modules.md [hunk 1]
```diff
@@ -48,6 +48,8 @@ Open the Visual Studio solution in the `windows` folder and add the new files di
 | `ReactConstant`         | Specifies a field or property that represents a constant. |
 | `ReactConstantProvider` | Specifies a method that provides a set of constants.      |
 | `ReactEvent`            | Specifies a field or property that represents an event.   |
+| `ReactInit`             | Specifies a class initialization method.                  |
+| `ReactFunction`         | Specifies a field that helps calling a function.          |
```
Comment (asklar) [inline docs/native-modules.md:52]: what does "a field that helps calling a function" mean? when would it be used? how is react function different from react method?

Comment (chrisglein) [reply docs/native-modules.md:52]: From the [header](https://github.com/microsoft/react-native-windows/blob/0dfe9decb4390c4b77e0837e69c4a25878034960/vnext/Microsoft.ReactNative.Cxx/NativeModules.h) it looks like a std::function. Or in C# I assume it's a Func/Action/whatever? Anyway, I wasn't sure how much detail to go into. Really these should all link to a page that has the full detail that's in that header.

Comment (chrisglein) [reply docs/native-modules.md:52]: The name is a bit confusing. Many engineers are going to use the terms method and function interchangeably. Maybe this should have been `ReactDelegate`?

Comment (vmoroz) [inline docs/native-modules.md:52]: >function [](http://example.com/codeflow?start=67&length=8)

We can say "helps calling a JavaScript function"

File: docs/native-modules.md [hunk 2]
```diff
@@ -48,6 +48,8 @@ Open the Visual Studio solution in the `windows` folder and add the new files di
 | `ReactConstant`         | Specifies a field or property that represents a constant. |
 | `ReactConstantProvider` | Specifies a method that provides a set of constants.      |
 | `ReactEvent`            | Specifies a field or property that represents an event.   |
+| `ReactInit`             | Specifies a class initialization method.                  |
```
Comment (vmoroz) [inline docs/native-modules.md:51]: >class [](http://example.com/codeflow?start=40&length=5)

module

**Review by vmoroz — APPROVED** [review]
:shipit:

**Comment by chrisglein** [comment]
> Is the 959 files changed by design?

Definitely nope. Refresh.

File: docs/native-modules.md [hunk 3]
```diff
@@ -51,7 +51,9 @@ Open the Visual Studio solution in the `windows` folder and add the new files di
 | `ReactConstant`         | Specifies a field or property that represents a constant. |
 | `ReactConstantProvider` | Specifies a method that provides a set of constants.      |
 | `ReactEvent`            | Specifies a field or property that represents an event.   |
-| `ReactStruct`           | Specifies a `struct` that can be used in native methods.    |
+| `ReactStruct`           | Specifies a `struct` that can be used in native methods.  |
+| `ReactInit`             | Specifies a class initialization module.                  |
+| `ReactFunction`         | Specifies a field that helps calling a JavaScript function. |
```
Comment (rectified95) [inline docs/native-modules.md:56]: Just by looking at this sentence I don't know what `ReactFunction` does :(
In NativeModules.h we have 'REACT_FUNCTION annotates a field that helps calling a JavaScript function.' so basically the same thing.
REACT_METHOD specifies the exported function to JS form native, what does this one do though?

Comment (chrisglein) [reply docs/native-modules.md:56]: I agree, that doesn't really help differentiate. I took that string from a previous comment from @vmoroz. Vladimir, is there a better way to capture the difference between REACT_METHOD and REACT_FUNCTION?

Comment (jonthysell) [inline docs/native-modules.md:56]:
```suggestion
| `ReactFunction`         | Specifies a JavaScript function that you want exposed to your native code. |

```

File: docs/native-modules.md [hunk 4]
```diff
@@ -173,6 +175,8 @@ The `Microsoft.ReactNative.Managed.ReactPackageProvider` is a convenience that m
 | `REACT_CONSTANTPROVIDER` | Specifies a method that provides a set of constants.      |
 | `REACT_EVENT`            | Specifies a field or property that represents an event.   |
 | `REACT_STRUCT`           | Specifies a `struct` that can be used in native methods (don't nest the definition inside `REACT_MODULE`).    |
+| `REACT_INIT`             | Specifies a class initialization module.                  |
+| `REACT_FUNCTION`         | Specifies a field that helps calling a JavaScript function.          |
```
Comment (jonthysell) [inline docs/native-modules.md:179]:
```suggestion
| `ReactFunction`         | Specifies a JavaScript function that you want exposed to your native code. |
```

File: website/versioned_docs/version-0.62/native-modules.md [hunk 5]
```diff
@@ -238,6 +240,8 @@ For events, you'll see that we created an instance of `NativeEventEmitter` passi
 | `REACT_CONSTANT`         | Specifies a field or property that represents a constant. |
 | `REACT_CONSTANTPROVIDER` | Specifies a method that provides a set of constants.      |
 | `REACT_EVENT`            | Specifies a field or property that represents an event.   |
+| `REACT_INIT`             | Specifies a class initialization module.                  |
+| `REACT_FUNCTION`         | Specifies a field that helps calling a JavaScript function.          |
```
Comment (jonthysell) [inline website/versioned_docs/version-0.62/native-modules.md:244]:
```suggestion
| `ReactFunction`         | Specifies a JavaScript function that you want exposed to your native code. |
```

File: website/versioned_docs/version-0.63/native-modules.md [hunk 6]
```diff
@@ -49,6 +49,8 @@ Open the Visual Studio solution in the `windows` folder and add the new files di
 | `ReactConstant`         | Specifies a field or property that represents a constant. |
 | `ReactConstantProvider` | Specifies a method that provides a set of constants.      |
 | `ReactEvent`            | Specifies a field or property that represents an event.   |
+| `ReactInit`             | Specifies a class initialization module.                  |
+| `ReactFunction`         | Specifies a field that helps calling a JavaScript function. |
```
Comment (jonthysell) [inline website/versioned_docs/version-0.63/native-modules.md:53]:
```suggestion
| `ReactFunction`         | Specifies a JavaScript function that you want exposed to your native code. |
```

File: website/versioned_docs/version-0.63/native-modules.md [hunk 7]
```diff
@@ -238,6 +240,8 @@ For events, you'll see that we created an instance of `NativeEventEmitter` passi
 | `REACT_CONSTANT`         | Specifies a field or property that represents a constant. |
 | `REACT_CONSTANTPROVIDER` | Specifies a method that provides a set of constants.      |
 | `REACT_EVENT`            | Specifies a field or property that represents an event.   |
+| `REACT_INIT`             | Specifies a class initialization module.                  |
+| `REACT_FUNCTION`         | Specifies a field that helps calling a JavaScript function.          |
```
Comment (jonthysell) [inline website/versioned_docs/version-0.63/native-modules.md:244]:
```suggestion
| `ReactFunction`         | Specifies a JavaScript function that you want exposed to your native code. |
```

File: website/versioned_docs/version-0.64/native-modules.md [hunk 8]
```diff
@@ -51,6 +51,8 @@ Open the Visual Studio solution in the `windows` folder and add the new files di
 | `ReactConstantProvider` | Specifies a method that provides a set of constants.      |
 | `ReactEvent`            | Specifies a field or property that represents an event.   |
 | `ReactStruct`           | Specifies a `struct` that can be used in native methods.    |
+| `ReactInit`             | Specifies a class initialization module.                  |
+| `ReactFunction`         | Specifies a field that helps calling a JavaScript function. |
```
Comment (jonthysell) [inline website/versioned_docs/version-0.64/native-modules.md:55]:
```suggestion
| `ReactFunction`         | Specifies a JavaScript function that you want exposed to your native code. |
```

File: website/versioned_docs/version-0.64/native-modules.md [hunk 9]
```diff
@@ -241,6 +243,8 @@ For events, you'll see that we created an instance of `NativeEventEmitter` passi
 | `REACT_CONSTANTPROVIDER` | Specifies a method that provides a set of constants.      |
 | `REACT_EVENT`            | Specifies a field or property that represents an event.   |
 | `REACT_STRUCT`           | Specifies a `struct` that can be used in native methods (don't nest the definition inside `REACT_MODULE`).    |
+| `REACT_INIT`             | Specifies a class initialization module.                  |
+| `REACT_FUNCTION`         | Specifies a field that helps calling a JavaScript function.          |
```
Comment (jonthysell) [inline website/versioned_docs/version-0.64/native-modules.md:247]:
```suggestion
| `ReactFunction`         | Specifies a JavaScript function that you want exposed to your native code. |
```

**Review by jonthysell — APPROVED** [review]

**Pull request merged**